lua/net.lua

```diff
@@ -1,17 +1,61 @@
 local gatekeeperc = require("gatekeeperc")
+local ffi = require("ffi")
+local ifaces = require("if_map")
 
 local M = {}
 
+function init_pci_addrs(iface, ports)
+	local pci_strs = ffi.new("const char *[" .. #ports .. "]")
+	for i, v in ipairs(ports) do
+		pci_strs[i - 1] = ifaces[v]
+	end
+	iface.pci_addrs = gatekeeperc.lua_init_pci_addrs(pci_strs, #ports)
+end
+
 -- Function that sets up the network.
 function M.setup_block()
 
 	-- Init the network configuration structure.
 	local conf = gatekeeperc.get_net_conf()
-	conf.num_rx_queues = 1
-	conf.num_tx_queues = 1
+
+	-- Change these parameters to configure the network.
+	local front_ports = {"enp131s0f0"}
+	local back_ports = {"enp133s0f0"}
+	conf.num_rx_queues = 2
+	conf.num_tx_queues = 2
+
+	-- Code below this point should not need to be changed.
+	conf.front.num_ports = #front_ports
+	conf.front.name = gatekeeperc.lua_init_iface_name("front")
```

Comment:
Thanks, Qiaobin. Could you take a look at the most recent push? I put all of the interface initialization that Lua can do into one C function, which I think looks much better.

Although, I still think we need the separate file `generate_if_map.c`, since the point is to be able to run it while the interfaces are still bound to the kernel driver (during the setup stage). That way, the map is built before the interfaces are changed to the DPDK driver, because at that point we won't be able to easily get them. By the time `gatekeeper` is run, the names will be lost.